modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc

```diff
@@ -606,39 +625,64 @@ void PointPillars::DoInference(const float* in_points_array,
+  GPU_CHECK(cudaMemcpyAsync(pfe_buffers_[2], dev_pillar_coors_,
+                            kMaxNumPillars * 4 * sizeof(float),
                             cudaMemcpyDeviceToDevice, stream));
-  pfe_context_->enqueueV2(pfe_buffers_, stream, nullptr);
 
-  GPU_CHECK(
-      cudaMemset(dev_scattered_feature_, 0, kRpnInputSize * sizeof(float)));
-  scatter_cuda_ptr_->DoScatterCuda(
-      host_pillar_count_[0], dev_x_coors_, dev_y_coors_,
-      reinterpret_cast<float*>(pfe_buffers_[1]), dev_scattered_feature_);
+  torch::Tensor tensor_pillar_point_feature = torch::from_blob(
+      pfe_buffers_[0],
+      {kMaxNumPillars, kMaxNumPointsPerPillar, kNumPointFeature}, torch::kCUDA);
+  torch::Tensor tensor_num_points_per_pillar =
+      torch::from_blob(pfe_buffers_[1], {kMaxNumPillars}, torch::kCUDA);
+  torch::Tensor tensor_pillar_coors =
+      torch::from_blob(pfe_buffers_[2], {kMaxNumPillars, 4}, torch::kCUDA);
 
-  GPU_CHECK(cudaMemcpyAsync(rpn_buffers_[0], dev_scattered_feature_,
-                            kBatchSize * kRpnInputSize * sizeof(float),
-                            cudaMemcpyDeviceToDevice, stream));
-  rpn_context_->enqueueV2(rpn_buffers_, stream, nullptr);
+  torch::Device device(device_type_, device_id_);
+  tensor_pillar_point_feature.to(device);
+  tensor_num_points_per_pillar.to(device);
+  tensor_pillar_coors.to(device);
+
+  torch::Tensor scattered_batch_size = torch::ones(1);
+  scattered_batch_size.to(device);
+
+  auto pfe_output =
+      pfe_net_
+          .forward({tensor_pillar_point_feature, tensor_num_points_per_pillar,
+                    tensor_pillar_coors})
+          .toTensor();
+
+  auto scattered_feature =
+      scattered_net_
+          .forward({pfe_output, tensor_pillar_coors, scattered_batch_size})
+          .toTensor();
+
+  auto backbone_feature = backbone_net_.forward({scattered_feature});
+
+  auto fpn_feature = fpn_net_.forward({backbone_feature});
+
+  auto bbox_head_output = bbox_head_net_.forward({fpn_feature}).toTuple();
+
+  auto cls_score = bbox_head_output->elements()[0].toTensor();
+  auto bbox_pred = bbox_head_output->elements()[1].toTensor();
+  auto dir_cls_preds = bbox_head_output->elements()[2].toTensor();
 
   GPU_CHECK(cudaMemset(dev_filter_count_, 0, sizeof(int)));
   postprocess_cuda_ptr_->DoPostprocessCuda(
-      reinterpret_cast<float*>(rpn_buffers_[1]),
-      reinterpret_cast<float*>(rpn_buffers_[2]),
-      reinterpret_cast<float*>(rpn_buffers_[3]), dev_anchor_mask_,
-      dev_anchors_px_, dev_anchors_py_, dev_anchors_pz_, dev_anchors_dx_,
-      dev_anchors_dy_, dev_anchors_dz_, dev_anchors_ro_, dev_filtered_box_,
-      dev_filtered_score_, dev_filtered_label_, dev_filtered_dir_,
-      dev_box_for_nms_, dev_filter_count_, out_detections, out_labels);
+      reinterpret_cast<float*>(bbox_pred.data_ptr<float>()),
+      reinterpret_cast<float*>(cls_score.data_ptr<float>()),
+      reinterpret_cast<float*>(dir_cls_preds.data_ptr<float>()),
```

Comment:
As `data_ptr<float>()` is already `float*` type, there is no need for `reinterpret_cast<float*>` .